python/cudf/cudf/core/groupby/groupby.py

```diff
@@ -185,7 +186,7 @@ def nth(self, n):
         Return the nth row from each group.
         """
         result = self.agg(lambda x: x.nth(n))
-        sizes = self.size()
+        sizes = self.size().sort_index()
```

Comment:
I'm still not clear as to why we need to call `sort_index` here. `result` won't be sorted by the index above so the generated boolean mask wouldn't be valid here.